### PR TITLE
feat(kafka): use kafka header format 'both' by default

### PR DESCRIPTION
**Pull request by basti1302**
This starts phase 1 of the Kafka header migration.

See https://www.ibm.com/docs/en/instana-observability/current?topic=references-kafka-header-migration for more information.

# Timing

**This should go live for customers in early October, but not before October 1.** 

This PR is currently set to draft so we do not merge and release it too early. I would appreciate a review, though.

### CODE REVIEW

File: instrumentation/instasarama/consumer_group_handler_test.go
Comment (basti1302) [inline instrumentation/instasarama/consumer_group_handler_test.go:26]: I'm entirely not sure about any of the changes in the tests for the *consumers*. Apparently we add headers to messages on the consumer side when no headers have been present in the original incoming message? That concept was new to me and a bit suprising. But from what I understand it is used so that the customers' applications can read the headers and continue the trace from the trace ID/span ID in the headers, is that correct?

If that understanding is correct, the header names and their contents are actually part of our public API in go and we need to make customers aware of this change, so they can update their application code somewhere between October 2022 and October 2023.

File: instrumentation/instasarama/consumer_group_handler_test.go
Comment (basti1302) [inline instrumentation/instasarama/consumer_group_handler_test.go:137]: I'm entirely not sure about any of the changes in the tests for the *consumers*, in particular the cases where the original message has no headers and then we check in the test that headers have been added _at the consumer side_? That is different from what our other tracers do, which only add headers at the producer side, so this concept was new to me and a bit suprising. But from what I understand it is done so that the customers' applications can read the headers and continue the trace from the trace ID/span ID in the headers, is that correct?

If that understanding is correct, the header names and their contents are actually part of our public API in go and we need to make customers aware of this change, so they can update their application code somewhere between October 2022 and October 2023.

File: instrumentation/instasarama/consumer_group_handler_test.go
Comment (basti1302) [inline instrumentation/instasarama/consumer_group_handler_test.go:37]: Not sure if this is a valid way to test this, please double check.

Comment (willianpc) [reply instrumentation/instasarama/consumer_group_handler_test.go:37]: Looks fine for me.

**Review by willianpc — COMMENTED** [review]
LGTM. Waiting for a review from the others.

File: instrumentation/instasarama/README.md
Comment (basti1302) [reply instrumentation/instasarama/README.md:140]: Thanks, I meant to write "will add ... to Kafka messages". The second "to" should not be there, I'll fix that.

Comment (basti1302) [reply instrumentation/instasarama/README.md]: fixed

File: instrumentation/instasarama/README.md
Comment (sanojsubran) [inline instrumentation/instasarama/README.md:126]: `Suggestion`: Since v1.2.0, the instrumentation supports Instana's trace correlation headers in both binary (legacy) and string (new) formats.

File: instrumentation/instasarama/async_producer_test.go
Comment (sanojsubran) [inline instrumentation/instasarama/async_producer_test.go:20]: `nitpicking` `opinionated` : `text` or `plaintext` format would have been better than `string` format. Just to back up my unnecessary argument, [binary strings](https://www.ibm.com/docs/en/db2/11.5?topic=list-binary-strings) are a thing in the IBM universe. ;)

File: instrumentation/instasarama/README.md
Comment (basti1302) [reply instrumentation/instasarama/README.md]: Thanks, updated.

**Comment by basti1302** [comment]
@willianpc @sanojsubran I've applied your readme suggestions, please re-review or approve if there are no further comments.

**Review by willianpc — APPROVED** [review]
LGTM

**Review by sanojsubran — APPROVED** [review]
LGTM

**Comment by basti1302** [comment]
@instana/eng-go This can now be merged and released some time next week.

**Pull request merged**
